pkg/services/datasources/service/legacy.go

```diff
@@ -0,0 +1,90 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	data "github.com/grafana/grafana-plugin-sdk-go/experimental/apis/data/v0alpha1"
+
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+// LegacyDataSourceRetriever supports finding a reference to datasources using the name or internal ID
+type LegacyDataSourceLookup interface {
+	// Find the UID from either the name or internal id
+	// NOTE the orgID will be fetched from the context
+	GetDataSourceFromDeprecatedFields(ctx context.Context, name string, id int64) (*data.DataSourceRef, error)
```

Comment:
@marefr -- I know we could use the `DataSourceRetriever` interface, but with this approach we can isolate all the actual configs.  Usage of this also does not need to clear a cache since nothing new should produce dashboards that need to lookup by name or id.